auction-server/src/state.rs

```diff
@@ -578,11 +610,86 @@ impl ChainStoreSvm {
         }
     }
 
+    pub async fn get_tx_receiver(&self) -> anyhow::Result<()> {
+        let ws_client = self.get_ws_client().await?;
+        let (mut stream, _) = ws_client
+            .logs_subscribe(
+                RpcTransactionLogsFilter::Mentions(vec![self
+                    .config
+                    .express_relay_program_id
+                    .to_string()]),
+                RpcTransactionLogsConfig {
+                    commitment: Some(CommitmentConfig::confirmed()),
+                },
+            )
+            .await
+            .unwrap();
+        let mut exit_check_interval = tokio::time::interval(EXIT_CHECK_INTERVAL);
+        let chain_name = self.get_name().clone();
+        while !SHOULD_EXIT.load(Ordering::Acquire) {
+            tokio::select! {
+                rpc_log = stream.next() => {
+                    match rpc_log {
+                        None => return Err(anyhow!("Log trigger stream ended for chain: {}", &chain_name)),
+                        Some(rpc_log) => {
+                            tracing::debug!("New log trigger received for {} at {}: {:?}", &chain_name, OffsetDateTime::now_utc(), rpc_log.clone());
+                                if let Err(e) = self.log_sender.send(rpc_log) {
+                                    tracing::error!("Failed to send log to channel: {:?}", e);
+                                }
+                        }
+                    };
+                }
+                _ = exit_check_interval.tick() => {}
+            }
+        }
+        tracing::info!("Shutting down log listener svm...");
+        Ok(())
+    }
+
+    pub fn subscribe_logs(&self) -> Receiver<Response<RpcLogsResponse>> {
+        self.log_sender.subscribe()
+    }
+
     pub async fn send_transaction(
         &self,
         tx: &VersionedTransaction,
+        task_tracker: TaskTracker,
     ) -> solana_client::client_error::Result<Signature> {
-        self.tx_broadcaster_client.send_transaction(tx).await
+        let config = RpcSendTransactionConfig {
+            skip_preflight:       true,
+            preflight_commitment: None,
+            encoding:             None,
+            max_retries:          None,
```

Comment:
should we set this to 0 according to the guidelines?

Reply:
I'd just copy this one from Kamino
https://github.com/Kamino-Finance/orbit-link/blob/master/src/async_client/rpc_client.rs#L41

Reply:
finalized commitment is fine since we get the blockhash finalized as per https://docs.triton.one/chains/solana/sending-txs